FrEee/Scripts/AI/Vanguard/MinistryOfColonization.csx

```diff
@@ -0,0 +1,296 @@
+            if (distance == int.MaxValue)
+                continue;
+            //work out the priority. 
+
+            CalculateScore(colonyPlan);
+            colonyPlan.AssignId(); 
+            PlanManager.CurrentPlans.ColonizationPlans.Add(colonyPlan); 
+            
+        }
+        
+    }
+
+    /// <summary>
+    /// Processes the current colonization to order it to move. 
+    /// </summary>
+    public void ProcessColonization()
+    {
+        //If the plan is completed, say so, or someone else got there first? 
+        foreach (var plan in PlanManager.CurrentPlans.ColonizationPlans.Where(x => x.ColonyShipDespatched && x.Planet.HasColony))
+        {
+            plan.IsComplete = true;
+            if (plan.ShipBuildPlan != null)
+            {
+                plan.ShipBuildPlan.IsComplete = true;
+            }
+            if (plan.Ship != null)
+                plan.Ship.Orders.Clear();
+        }
+
+        //Assign unassigned colony ships to the colony plan
+        var unassignedShips = Empire.OwnedSpaceObjects.OfType<Ship>().Where(x =>
+        (x.HasAbility("Colonize Planet - Rock") || x.HasAbility("Colonize Planet - Ice") || x.HasAbility("Colonize Planet - Gas"))
+        && x.IsIdle && !PlanManager.CurrentPlans.ColonizationPlans.Select(x => x.ShipId).Contains(x.ID)); 
+        foreach(var ship in unassignedShips)
+        {
+            UpdateUnstartedPlans(); 
+            ColonizationPlan currentPlan = null;
+            int currentPriority = int.MinValue; 
+            foreach(var plan in PlanManager.CurrentPlans.ColonizationPlans.Where(x => x.Ship == null 
+                && (x.ShipBuildPlanId == null || !x.ShipBuildPlan.SentOrder)  && ship.HasAbility(x.Planet.ColonizationAbilityName)))
+            {
+                var distance = Pathfinder.EstimateDistance(ship.Sector, plan.Planet.Sector, Empire);
+                var potentialPriority =  plan.Priority - (distance * 100); 
+
+                if (potentialPriority > currentPriority)
+                {
+                    currentPriority = potentialPriority;
+                    currentPlan = plan; 
+                }
+            }
+            if (currentPlan == null)
+                continue; 
+
+            currentPlan.Ship = ship;
+            currentPlan.AwaitingBuild = false;
+            if (currentPlan.ShipBuildPlan != null)
+            {
+                currentPlan.ShipBuildPlan.IsComplete = true;
+            }
+        }
+
+
+
+        var plansToProcess = PlanManager.CurrentPlans.ColonizationPlans.Where(x => !x.ColonyShipDespatched && x.Ship != null); 
+        foreach(var plan in plansToProcess)
+        {
+            ///someone got here first?! Cancel the plan. 
+            if (plan.Planet.Colony != null)
+            {
+                plan.IsComplete = true; 
+                continue; 
+            }
+            var ship = plan.Ship; 
+            //load some population. 
+            if (ship.Cargo.Population.Count < 1)
```

Comment:
Is the population loading logic copied from the GUI when the player issues a colonization order? It looks kind of familiar! Might want to refactor it out into a utility method if that's the case.

Reply:
yep. It is copied. That is the only place it is needed currently though.

Reply:
So it's used in two places, the GUI colonization orders and here in the AI? Would make sense to refactor it into a method in that case.

Reply:
Oh well, not a big deal, we can merge in this PR and refactor later if need be...